include/ginkgo/core/solver/multigrid.hpp

```diff
@@ -537,6 +607,8 @@ class Multigrid : public EnableLinOp<Multigrid> {
     std::shared_ptr<const LinOp> coarsest_solver_{};
     std::function<size_type(const size_type, const LinOp*)> level_selector_;
     std::function<size_type(const size_type, const LinOp*)> solver_selector_;
+    mutable multigrid::MultigridState state;
```

Comment:
Instead of exposing this like this to the public, can't we use a `std::unique_ptr` of a forward-declared `MultigridState` and keep this `MultigridState` thing hidden from the public interface?

Reply:
Thanks for suggestion!